src/CarbonAware.CLI/src/Commands/Emissions/EmissionsCommand.cs

```diff
@@ -0,0 +1,45 @@
+using CarbonAware.Aggregators.CarbonAware;
+using CarbonAware.CLI.Common;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Text.Json;
+
+namespace CarbonAware.CLI.Commands.Emissions;
+
+class EmissionsCommand : Command
+{
+    private Option<string[]> _requiredLocationOption = CommonOptions.RequiredLocationOption;
+    private Option<DateTimeOffset?> _startTimeOption = CommonOptions.StartTimeOption;
+    private Option<DateTimeOffset?> _endTimeOption = CommonOptions.EndTimeOption;
+    public EmissionsCommand() : base("emissions", LocalizableStrings.EmissionsCommandDescription)
+    {
+        AddOption(_requiredLocationOption);
+        AddOption(_startTimeOption);
+        AddOption(_endTimeOption);
+        this.SetHandler(this.Run);
+    }
+
+    private async Task Run(InvocationContext context)
+    {
+        // Get aggregator via DI.
+        var serviceProvider = context.BindingContext.GetService(typeof(IServiceProvider)) as IServiceProvider ?? throw new NullReferenceException("ServiceProvider not found");
```

Comment:
hmm, this basically breaks the whole DI idea. It should be injected as part of the constructor.

Reply:
As we discussed out-of-band: doing DI in the handler is the [recommended approach](https://docs.microsoft.com/en-us/dotnet/standard/commandline/dependency-injection) because:

> - Command-line apps are often short-lived processes, in which startup cost can have a noticeable impact on performance. Optimizing performance is particularly important when tab completions have to be calculated. Command-line apps are unlike Web and GUI apps, which tend to be relatively long-lived processes. Unnecessary startup time is not appropriate for short-lived processes.
> - When a command-line app that has multiple subcommands is run, only one of those subcommands will be executed. If an app configures dependencies for the subcommands that don't run, it needlessly degrades performance.

I think we should revisit this discussion after we have implemented the rest of the commands and consider the right pattern for DI when there is enough code to see such patterns :)